lib/src/compile.ts

```diff
@@ -87,10 +90,19 @@ function newCompileStringRequest(
   input.setSource(source);
   input.setSyntax(utils.protofySyntax(options?.syntax ?? 'scss'));
 
-  if (options?.url) input.setUrl(options.url.toString());
+  const url = options?.url?.toString();
+  if (url && url !== legacyImporterProtocol) {
```

Comment:
The special value `legacyImporterProtocol` means this is from Legacy API that even if it does not set a `url` it should behave as if it has one down below when setting the importer.